### PR TITLE
imgui: 1.90.6 -> 1.91.4

### LINKED ISSUE #349883: Update request: imgui 1.90.6 → 1.91.4
- Package name: imgui
- Latest released version: 1.91.4
- Current version on the unstable channel: 1.90.6
- Current version on the stable/release channel: 1.90.6
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@WolfangAukang

https://github.com/ocornut/imgui/releases/tag/v1.91.4

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jfvillablanca**
https://github.com/ocornut/imgui/releases/tag/v1.91.4

## Things done
Closes #349883

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/imgui/default.nix [hunk 1]
```diff
@@ -70,7 +70,7 @@ stdenv.mkDerivation rec {
     owner = "ocornut";
     repo = "imgui";
     rev = "v${version}";
-    sha256 = "sha256-FSob6FPfg0tF0n72twA5/moLvEaB251BPkIDJUXhYbg=";
+    sha256 = "sha256-6j4keBOAzbBDsV0+R4zTNlsltxz2dJDGI43UIrHXDNM=";
```
Comment (khaneliman) [inline pkgs/development/libraries/imgui/default.nix:73]:
```suggestion
    hash = "sha256-6j4keBOAzbBDsV0+R4zTNlsltxz2dJDGI43UIrHXDNM=";
```

File: pkgs/development/libraries/imgui/default.nix [hunk 2]
```diff
@@ -70,7 +70,7 @@ stdenv.mkDerivation rec {
     owner = "ocornut";
     repo = "imgui";
     rev = "v${version}";
```
Comment (khaneliman) [inline pkgs/development/libraries/imgui/default.nix:72]:
```suggestion
    rev = "refs/tags/v${version}";
```
If we're in here doing a manual update.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349885`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>imgui</li>
    <li>imgui.lib</li>
    <li>imnodes</li>
    <li>imnodes.dev</li>
    <li>implot</li>
    <li>imtui</li>
    <li>sony-headphones-client</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>imgui</li>
    <li>imgui.lib</li>
    <li>imnodes</li>
    <li>imnodes.dev</li>
    <li>implot</li>
    <li>imtui</li>
    <li>sony-headphones-client</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>imgui</li>
    <li>imgui.lib</li>
    <li>imnodes</li>
    <li>imnodes.dev</li>
    <li>implot</li>
    <li>imtui</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>imgui</li>
    <li>imgui.lib</li>
    <li>imnodes</li>
    <li>imnodes.dev</li>
    <li>implot</li>
    <li>imtui</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM

**Review by NickCao — APPROVED** [review]

**Pull request merged**